pages/docs/manual/latest/variant.mdx

```diff
@@ -193,44 +193,266 @@ let a1 = Warrior({score: 10.5})
 let a2 = Wizard("Joe")
 ```
 ```js
-var g1 = /* Hello */0;
-var g2 = /* Goodbye */1;
+var g1 = "Hello";
 
-var o1 = /* Good */0;
-var o2 = /* Error */{
+var g2 = "Goodbye";
+
+var o1 = "Good";
+
+var o2 = {
+  TAG: "Error",
   _0: "oops!"
 };
 
-var f1 = /* Child */0;
+var f1 = "Child";
+
 var f2 = {
-  TAG: /* Mom */0,
+  TAG: "Mom",
   _0: 30,
   _1: "Jane"
 };
+
 var f3 = {
-  TAG: /* Dad */1,
+  TAG: "Dad",
   _0: 32
 };
 
-var p1 = /* Teacher */0;
-var p2 = /* Student */{
+var p1 = "Teacher";
+
+var p2 = {
+  TAG: "Student",
   gpa: 99.5
 };
 
 var a1 = {
-  TAG: /* Warrior */0,
+  TAG: "Warrior",
   _0: {
     score: 10.5
   }
 };
+
 var a2 = {
-  TAG: /* Wizard */1,
+  TAG: "Wizard",
   _0: "Joe"
 };
 ```
 
 </CodeTab>
 
+## Tagged variants
+
+- The `@tag` attribute lets you customize the discriminator (default: `TAG`).
+- `@as` attributes control what each variant case is discriminated on (default: the variant case name as string).
+
+### Example: Binding to TypeScript enums
+
+```typescript
+// direction.ts
+/** Direction of the action. */
+enum Direction {
+  /** The direction is up. */
+  Up = "UP",
+
+  /** The direction is down. */
+  Down = "DOWN",
+
+  /** The direction is left. */
+  Left = "LEFT",
+
+  /** The direction is right. */
+  Right = "RIGHT",
+}
+
+export const myDirection = Direction.Up;
+```
+
+You can bind to the above enums like so:
+
+```rescript
+/** Direction of the action. */
+type direction =
+  | /** The direction is up. */
+  @as("UP")
+  Up
+
+  | /** The direction is down. */
+  @as("DOWN")
+  Down
+
+  | /** The direction is left. */
+  @as("LEFT")
+  Left
+
+  | /** The direction is right. */
+  @as("RIGHT")
+  Right
+
+@module("./direction.js") external myDirection: direction = "myDirection"
+```
+
+Now, this maps 100% to the TypeScript code, including letting us bring over the documentation strings so we get a nice editor experience.
+
+### String literals
+
+The same logic is easily applied to string literals from TypeScript, only here the benefit is even larger, because string literals have the same limitations in TypeScript that polymorphic variants have in ReScript.
+
+```typescript
+// direction.ts
+type direction = "UP" | "DOWN" | "LEFT" | "RIGHT";
+```
+
+There's no way to attach documentation strings to string literals in TypeScript, and you only get the actual value to interact with.
+
+## Untagged variants
```

Comment:
We should add a list of what can currently be unboxed (and why), and a note that it's possible that more things could be unboxed in the future.

@fhammerschmidt would you mind adding separate issues/tasks for that type of thing? I can probably add the actual list.

Reply:
And, we should add a small section on the "catch all" trick, and what kind of things that enable.